From 3d28fcc1e1f896eaa77765dcf355828c7968769b Mon Sep 17 00:00:00 2001
From: Alon Zakai <azakai@google.com>
Date: Mon, 18 Nov 2019 10:25:45 -0800
Subject: [PATCH] Fix #2430 properly

---
 src/wasm-features.h | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/wasm-features.h b/src/wasm-features.h
index 77759f9e8..c9496bb42 100644
--- a/src/wasm-features.h
+++ b/src/wasm-features.h
@@ -67,18 +67,18 @@ struct FeatureSet {
 
   bool isMVP() const { return features == MVP; }
   bool has(Feature f) { return (features & f) == f; }
-  bool hasAtomics() const { return bool(features & Atomics); }
-  bool hasMutableGlobals() const { return bool(features & MutableGlobals); }
-  bool hasTruncSat() const { return bool(features & TruncSat); }
-  bool hasSIMD() const { return bool(features & SIMD); }
-  bool hasBulkMemory() const { return bool(features & BulkMemory); }
-  bool hasSignExt() const { return bool(features & SignExt); }
+  bool hasAtomics() const { return (features & Atomics) != 0; }
+  bool hasMutableGlobals() const { return (features & MutableGlobals) != 0; }
+  bool hasTruncSat() const { return (features & TruncSat) != 0; }
+  bool hasSIMD() const { return (features & SIMD) != 0; }
+  bool hasBulkMemory() const { return (features & BulkMemory) != 0; }
+  bool hasSignExt() const { return (features & SignExt) != 0; }
   bool hasExceptionHandling() const {
-    return bool(features & ExceptionHandling);
+    return (features & ExceptionHandling) != 0;
   }
-  bool hasTailCall() const { return bool(features & TailCall); }
-  bool hasReferenceTypes() const { return bool(features & ReferenceTypes); }
-  bool hasAll() const { return bool(features & All); }
+  bool hasTailCall() const { return (features & TailCall) != 0; }
+  bool hasReferenceTypes() const { return (features & ReferenceTypes) != 0; }
+  bool hasAll() const { return (features & All) != 0; }
 
   void makeMVP() { features = MVP; }
   void set(Feature f, bool v = true) {
