From cf2bb4841e8315d1b6a633efcaa6672cd001f494 Mon Sep 17 00:00:00 2001
From: Alon Zakai <azakai@google.com>
Date: Tue, 12 Nov 2019 10:32:27 -0800
Subject: [PATCH 1/2] Cast to bool after bitwise operations, to avoid an MSVC
 warning. Fixes #2430

---
 src/wasm-features.h | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/wasm-features.h b/src/wasm-features.h
index a645c93a8..d51d334c9 100644
--- a/src/wasm-features.h
+++ b/src/wasm-features.h
@@ -67,16 +67,16 @@ struct FeatureSet {
 
   bool isMVP() const { return features == MVP; }
   bool has(Feature f) { return (features & f) == f; }
-  bool hasAtomics() const { return features & Atomics; }
-  bool hasMutableGlobals() const { return features & MutableGlobals; }
-  bool hasTruncSat() const { return features & TruncSat; }
-  bool hasSIMD() const { return features & SIMD; }
-  bool hasBulkMemory() const { return features & BulkMemory; }
-  bool hasSignExt() const { return features & SignExt; }
-  bool hasExceptionHandling() const { return features & ExceptionHandling; }
-  bool hasTailCall() const { return features & TailCall; }
-  bool hasReferenceTypes() const { return features & ReferenceTypes; }
-  bool hasAll() const { return features & All; }
+  bool hasAtomics() const { return bool(features & Atomics); }
+  bool hasMutableGlobals() const { return bool(features & MutableGlobals); }
+  bool hasTruncSat() const { return bool(features & TruncSat); }
+  bool hasSIMD() const { return bool(features & SIMD); }
+  bool hasBulkMemory() const { return bool(features & BulkMemory); }
+  bool hasSignExt() const { return bool(features & SignExt); }
+  bool hasExceptionHandling() const { return bool(features & ExceptionHandling); }
+  bool hasTailCall() const { return bool(features & TailCall); }
+  bool hasReferenceTypes() const { return bool(features & ReferenceTypes); }
+  bool hasAll() const { return bool(features & All); }
 
   void makeMVP() { features = MVP; }
   void set(Feature f, bool v = true) {

From 1a08ace0e6fd3f3574195aaffe444244c8207648 Mon Sep 17 00:00:00 2001
From: Alon Zakai <azakai@google.com>
Date: Tue, 12 Nov 2019 10:32:44 -0800
Subject: [PATCH 2/2] format

---
 src/wasm-features.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/wasm-features.h b/src/wasm-features.h
index d51d334c9..77759f9e8 100644
--- a/src/wasm-features.h
+++ b/src/wasm-features.h
@@ -73,7 +73,9 @@ struct FeatureSet {
   bool hasSIMD() const { return bool(features & SIMD); }
   bool hasBulkMemory() const { return bool(features & BulkMemory); }
   bool hasSignExt() const { return bool(features & SignExt); }
-  bool hasExceptionHandling() const { return bool(features & ExceptionHandling); }
+  bool hasExceptionHandling() const {
+    return bool(features & ExceptionHandling);
+  }
   bool hasTailCall() const { return bool(features & TailCall); }
   bool hasReferenceTypes() const { return bool(features & ReferenceTypes); }
   bool hasAll() const { return bool(features & All); }
